src/redux/pokemons/pokemons.reducer.js

```diff
@@ -0,0 +1,70 @@
+import PokemonsActionTypes from './pokemons.types';
+
+const INITIAL_STATE = {
+  freePokemons: null,
+  capturedPokemons: null,
+  loading: false
+};
+
+const pokemonsReducer = (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case PokemonsActionTypes.GET_POKEMON_DATA:
+      return {
+        ...state,
+        loading: false,
+        freePokemons: action.payload.freePokemons,
+        capturedPokemons: action.payload.capturedPokemons
+      };
+    case PokemonsActionTypes.GET_POKEMON_DATA_ASYNC:
+      return {
+        ...state,
+        loading: true
+      };
+    case PokemonsActionTypes.CATCH_POKEMON:
+      return {
+        ...state,
+        freePokemons: state.freePokemons.filter(
+          pokemon => pokemon.id !== action.payload.id
+        ),
+        capturedPokemons: [
+          ...state.capturedPokemons.slice(
+            0,
+            action.payload.id === 0 ? null : action.payload.id - 1
+          ),
+          {
+            ...action.payload,
+            status: 'captured',
+            captureDate: new Date()

```

Comment:
you shoudn't do something like that in a reducer. Reducer should be a pure function - it should use just input parameters. no more no less. here you call constructor of Date object - it's side effect, because it's some logic that will return each time different date objects and you can get different results from function with the same input params. it conflicts with definition of a pure function. this logic could be placed in action creator or middleware (I mean in sagas in your case)

Reply:
Okay, do not keep any functions in a reducer which might cause any side effects. But its fine about directly changing fields, like `status: 'captured'` above, right?

Reply:
it's possible, but I don't recommend you it)